project_event/tests/test_project_event_task.py

```diff
@@ -527,3 +527,37 @@ def test_220_get_parent_project_id(self):
             self.task_2.get_parent_project_id(),
             self.project_2.id
         )
+
+    def test_230_is_type_task(self):
+        check_task_type_vals = {
+            'name': 'Sample Task for checking task type',
+            'activity_task_type': 'task',
+            'partner_id': self.project_1.partner_id.id,
+            'room_id': self.room_1.id,
+            'date_start': fields.Datetime.to_string(datetime.today()),
+            'date_end': fields.Datetime.to_string(datetime.today() +
+                                                  timedelta(hours=4)),
+        }
+        task = self.Tasks.create(check_task_type_vals)
+        self.assertEqual(task.name, 'Sample Task for checking task type')
+        self.assertEqual(task.activity_task_type, 'task')
+        check_task_type = task.is_type_task()
+        self.assertEqual(check_task_type, True)
+
+    def test_240_check_task_state(self):
+        check_task_state_vals = {
+            'name': 'Sample Task for checking task state',
+            'activity_task_type': 'task',
+            'partner_id': self.project_1.partner_id.id,
+            'room_id': self.room_1.id,
+            'task_state': 'option',
+            'date_start': fields.Datetime.to_string(datetime.today()),
+            'date_end': fields.Datetime.to_string(datetime.today() +
+                                                  timedelta(hours=4)),
+        }
+        task = self.Tasks.create(check_task_state_vals)
+        self.assertEqual(task.name, 'Sample Task for checking task state')
+        task.action_option()
+        self.assertEqual(task.task_state, 'option')
+        check_task_state = task.check_task_state(task.task_state)
+        self.assertEqual(check_task_state, True)
```

Comment:
simply check these 2 cases:
self.assertTrue(self.task1.check_task_state('draft'))
self.assertFalse(self.task1.check_task_state('......any state that is not in the test domain'))